File: codeforces/172/D.py
from math import sqrt
a, n = map(int, raw_input().split())
b = a + n
m = [1] * b
for i in xrange(2, int(sqrt(b)) + 1):
  ii = i * i
  for j in xrange(0, b, ii):
    m[j] = ii
m = map(lambda (x, y): x / y, zip(range(b), m))
print sum(m[a:b])

File: codeforces/172/D.scala
import java.util._
import scala.math._

object Main {
    val in = new Scanner(System.in)

    def main(args: Array[String]) {
        val a = in.nextInt()
        val n = in.nextInt()
        val b = a + n
        val x = Array.fill[Long](b)(1)
        for (i <- 1 to sqrt(b).toInt) {
            val ii = i * i
            for (j <- 0 until b by ii) {
                x(j) = ii
            }
        }
        println(x.zipWithIndex.drop(a).map{ case (x, y) => y / x }.sum)
    }
}

// vim: set ts=4 sw=4 et:
